crates/ide_assists/src/handlers/generate_getter.rs

```diff
@@ -154,7 +154,7 @@ fn useless_type_special_case(field_name: &str, field_ty: &ast::Type) -> Option<(
 }
 
 // FIXME: This should rely on semantic info.
-fn ty_ctor(ty: &ast::Type, ctor: &str) -> Option<String> {
+fn ty_ctor(ty: &String, ctor: &str) -> Option<String> {
```

Comment:
It's unfortunate that we go from `ast` to `String` here, but it's also true that ast is not correct either -- we want `hir::ty` here.

Reply:
OK, I don't have a clue how this should be done. Any suggestion?

Reply:
That's mostly me thinking aloud -- I don't think there's an easy way to do that right now, so the current impl is fine!

That said, if you already can see a way to change both call-sites to pass in a [Type](https://github.com/rust-analyzer/rust-analyzer/blob/6aeaceaf45508383c4664bc253fb21e48034d1bc/crates/hir/src/lib.rs#L2137-L2142), than we should do that (in a separate PR). 


In other words, it's clear that we want this to operate on semantic notion of type (`hir::Type`). The way to do it though is to carefully peruse exiting APIs and usages and make that possible by extending the API. It's not just "use this magic function". 

In general, our overall semantic APIs are a mess at the moment, that's why we have  https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Frust-analyzer/topic/Tree.20Based.20HIR discussion.

Reply:
Well, currently I'm not that familiar with AST and HIR. Guess I'll leave it like this for now.